### DISCUSSION TITLE
Wrong encode/decode after restart

### DISCUSSION THREAD

**Discussion by maxim092001** [gvolpe/pfps-shopping-cart#558] [Q&A]
Hi!

Thanks for an amazing book and project!

However, I've faced some strange issues while I was writing my own implementation. While my app is working users can create their account, get JWT token, log out and log in again. Username and encoded password saved into postgres. 

However, each time I restart my app encoder and decoder works differently, so for the same password, its encoded representation is different. With that login doesn't work.

![Screenshot 2022-02-15 at 22 35 35](https://user-images.githubusercontent.com/35135337/154135707-7556a341-2dd4-499e-a93c-758b0b1f12a7.png)

All users have the same password but were created before/after multiple restarts.

I found out that removing random generation for [ivBytes](https://github.com/maxim092001/stock-exchange-platform/blob/master/src/main/scala/auth/Crypto.scala#L27) fixed this problem, but I don't think this is the best solution.

Thanks!

P.S. you can find my implementation [here](https://github.com/maxim092001/stock-exchange-platform)

**Comment by gvolpe** [comment]
Hi,

That's a good finding, thanks for reporting it! 

That's definitely a bug in that (ugly-java-stuff) implementation and I will get it fixed at some point, though, that's only there as a reference and readers are encouraged to write their own. The interface is what matters most; the interpreter could be anything, really.
